model/task_lifecycle.go

```diff
@@ -1643,26 +1643,57 @@ func ClearAndResetStrandedHostTask(h *host.Host) error {
 	return nil
 }
 
-// ResetStaleTask fixes a task that has exceeded the heartbeat timeout by
-// marking the current task execution as finished and, if possible, a new
-// execution is created to restart the task.
+// ResetStaleTask fixes a task that has either exceeded the heartbeat timeout
+// or has been marked as aborted but was never ended by the agent.
```

Comment:
I think this second comment is a subset of the first one, which is that the heartbeat timed out because the agent has supposedly stopped running this task. For clarity of reading this comment, can we remove this mention and substantiate the later part (on L1651) with a comment on why an aborting task is treated as a special case for the heartbeat timeout?